# HG changeset patch
# User Olaf Till <i7tiol@t-online.de>
# Date 1517896809 -3600
#      Tue Feb 06 07:00:09 2018 +0100
# Branch stable
# Node ID cbe53b5ea5ce2eb2444392e09be2967f73b4253b
# Parent  72ea868ca63afafa0d5fe2f77d8a57f8a56643e4
check success of loading oct-file (bug #52977)

* oct-parse.in.yy (load_fcn_from_file): Ensure function returned from
load_oct is valid before using it.

diff --git a/libinterp/parse-tree/oct-parse.in.yy b/libinterp/parse-tree/oct-parse.in.yy
--- a/libinterp/parse-tree/oct-parse.in.yy
+++ b/libinterp/parse-tree/oct-parse.in.yy
@@ -4748,19 +4748,21 @@ namespace octave
     if (len > 4 && file.substr (len-4, len-1) == ".oct")
       {
         if (autoload && ! fcn_name.empty ())
           nm = fcn_name;
 
         octave_function *tmpfcn
           = dyn_loader.load_oct (nm, file, relative_lookup);
 
-        tmpfcn->stash_package_name (package_name);
-
-        retval = octave_value (tmpfcn);
+        if (tmpfcn)
+          {
+            tmpfcn->stash_package_name (package_name);
+            retval = octave_value (tmpfcn);
+          }
       }
     else if (len > 4 && file.substr (len-4, len-1) == ".mex")
       {
         // Temporarily load m-file version of mex-file, if it exists,
         // to get the help-string to use.
 
         std::string doc_string;
 
